common/component/postgresql/awsIAM/aws.go

```diff
@@ -0,0 +1,84 @@
+package awsiam
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	aws_config "github.com/aws/aws-sdk-go-v2/config"
+	aws_credentials "github.com/aws/aws-sdk-go-v2/credentials"
+	aws_auth "github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type AWSIAM struct {
+	// Ignored by metadata parser because included in built-in authentication profile
+	// access key to use for accessing postgresql.
+	AWSAccessKey string `json:"awsAccessKey" mapstructure:"awsAccessKey"`
+	// secret key to use for accessing postgresql.
+	AWSSecretKey string `json:"awsSecretKey" mapstructure:"awsSecretKey"`
+	// aws session token to use.
+	AWSSessionToken string `mapstructure:"awsSessionToken"`
+	// aws region in which postgresql should create resources.
+	AWSRegion string `mapstructure:"awsRegion"`
+}
+
+func GetAccessToken(ctx context.Context, pgCfg *pgx.ConnConfig, region, accessKey, secretKey string) (string, error) {
+	dbEndpoint := pgCfg.Host + ":" + strconv.Itoa(int(pgCfg.Port))
+	var authenticationToken string
+
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.Connecting.Go.html
+	// Default to load default config through aws credentials file (~/.aws/credentials)
+	awsCfg, err := aws_config.LoadDefaultConfig(ctx)
+	if err != nil {
```

Comment:
Can this error for any other reason other than the credential not being available at the well known locations? i.e. is this returned error typed which we need to select over.

Reply:
I see that there are 5 options to choose from when it comes to aws config pkg errors, and only one does not apply here, so I've updated accordingly.